code/obj/item/clothing/helmets.dm

```diff
@@ -123,53 +123,55 @@
+	var/image/visrItemImg = null
+	var/image/visrWornImg = null
 
 	New()
 		..()
-
-		visrImg = SafeGetOverlayImage("visor", src.icon, "spacemat-vis") // prep the world icon_state for building, is made later
-		fabrImg = SafeGetOverlayImage("helmet", src.icon, "spacemat")
-
-	proc/setupVisorMat(var/datum/material/V)
-		visr_material = copyMaterial(V) // in 99% of all calls this is redundant but just in case
-		if (visr_material)
-
-			var/prot = max(0, (5 - visr_material.getProperty("thermal")) * 5)
-			setProperty("coldprot", 10+prot)
-			setProperty("heatprot", 2+round(prot/2))
-
-			prot =  clamp(((visr_material.getProperty("chemical") - 4) * 10), 0, 35) // All crystals (assuming default chem value) will give 20 chemprot, same as normal helm.
-			setProperty("chemprot", prot)
-
-			prot = max(0, visr_material.getProperty("density") - 3) / 2
-			setProperty("meleeprot_head", 3 + prot) // even if soft visor, still gives some value
-
-		// overlay stuff
-
-		fabrImg.color = src.material
-		UpdateOverlays(fabrImg, "helmet")
-
-		visrImg.color = visr_material.color
-		UpdateOverlays(visrImg, "visor")
-
-
-	UpdateName()
-		if (visr_material && src.material)
-			name = "[visr_material]-visored [src.material] helmet"
-		else if (visr_material)
-			name = " [src.material] helmet"
+		// Prep the item overlays
+		fabrItemImg = SafeGetOverlayImage("item-helmet", src.icon, "spacemat")
+		visrItemImg = SafeGetOverlayImage("item-visor", src.icon, "spacemat-vis")
+		// Prep the worn overlays
+		fabrWornImg = SafeGetOverlayImage("worn-helmet", src.wear_image_icon, "spacemat")
+		visrWornImg = SafeGetOverlayImage("worn-visor", src.wear_image_icon, "spacemat-vis")
+
+	proc/set_custom_mats(datum/material/helmMat, datum/material/visrMat)
+		src.setMaterial(
+			helmMat,
+			FALSE, // We want to purely rely on the overlay colours
+		)
+		name = "[visrMat]-visored [helmMat] helmet"
+
+		// Setup the clothing stats based on material properties
+		var/prot = max(0, (5 - visrMat.getProperty("thermal")) * 5)
+		setProperty("coldprot", 10+prot)
+		setProperty("heatprot", 2+round(prot/2))
+		// All crystals (assuming default chem value) will give 20 chemprot, same as normal helm
+		prot =  clamp(((visrMat.getProperty("chemical") - 4) * 10), 0, 35)
+		setProperty("chemprot", prot)
+		 // Even if soft visor, still gives some value
+		prot = max(0, visrMat.getProperty("density") - 3) / 2
+		setProperty("meleeprot_head", 3 + prot)
+
+		// Setup item overlays
+		fabrItemImg.color = helmMat.color
+		visrItemImg.color = visrMat.color
+		UpdateOverlays(visrItemImg, "item-visor")
+		UpdateOverlays(fabrItemImg, "item-helmet")
+		// Setup worn overlays
+		fabrWornImg.color = helmMat.color
+		visrWornImg.color = visrMat.color
+		src.wear_image.overlays += fabrWornImg
+		src.wear_image.overlays += visrWornImg
+		// Add back the helmet texture since we overide the material apparance
+		if (helmMat.texture)
+			src.setTexture(helmMat.texture, helmMat.texture_blend, "material")
```

Comment:
This recolors the base helmet. However I wasn't able to figure out how to set the texture on the visor item / worn sprite overlays. If anyone has any pointers on how to add this, I'll gladly add it to this PR.